plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -189,46 +207,56 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
-                .put(Types.BINARY, StandardColumnMappings.varbinaryColumnMapping())
-                .put(Types.VARBINARY, StandardColumnMappings.varbinaryColumnMapping())
-                .put(Types.LONGVARBINARY, StandardColumnMappings.varbinaryColumnMapping())
-                .buildOrThrow();
-
-        ColumnMapping columnMap = standardColumnMappings.get(type);
-        if (columnMap != null) {
-            return Optional.of(columnMap);
-        }
-
-        final Map<String, ColumnMappingFunction> snowflakeColumnMappings = ImmutableMap.<String, ColumnMappingFunction>builder()
-                .put("time", handle -> Optional.of(timeColumnMapping(handle.getRequiredDecimalDigits())))
-                .put("timestampntz", handle -> Optional.of(timestampColumnMapping(handle.getRequiredDecimalDigits())))
-                .put("timestamptz", handle -> Optional.of(timestampTZColumnMapping(handle.getRequiredDecimalDigits())))
-                .put("date", handle -> Optional.of(ColumnMapping.longMapping(DateType.DATE, (resultSet, columnIndex) -> LocalDate.ofEpochDay(resultSet.getLong(columnIndex)).toEpochDay(), snowFlakeDateWriter())))
-                .put("varchar", handle -> Optional.of(varcharColumnMapping(handle.getRequiredColumnSize(), typeHandle.getCaseSensitivity())))
-                .put("number", handle -> {
-                    int decimalDigits = handle.getRequiredDecimalDigits();
-                    int precision = handle.getRequiredColumnSize() + Math.max(-decimalDigits, 0);
-                    if (precision > 38) {
-                        return Optional.empty();
-                    }
-                    return Optional.of(columnMappingPushdown(
-                            StandardColumnMappings.decimalColumnMapping(DecimalType.createDecimalType(precision, Math.max(decimalDigits, 0)), RoundingMode.UNNECESSARY)));
-                })
-                .buildOrThrow();
-
-        ColumnMappingFunction columnMappingFunction = snowflakeColumnMappings.get(jdbcTypeName);
-        if (columnMappingFunction != null) {
-            return columnMappingFunction.convert(typeHandle);
+        switch (type) {
+            case Types.BOOLEAN:
+                return Optional.of(booleanColumnMapping());
+            case Types.TINYINT:
+                return Optional.of(tinyintColumnMapping());
+            case Types.SMALLINT:
+                return Optional.of(smallintColumnMapping());
+            case Types.INTEGER:
+                return Optional.of(integerColumnMapping());
+            case Types.BIGINT:
+                return Optional.of(bigintColumnMapping());
+            case Types.REAL:
+                return Optional.of(realColumnMapping());
+            case Types.FLOAT:
+            case Types.DOUBLE:
+                return Optional.of(doubleColumnMapping());
+            case Types.NUMERIC:
+            case Types.DECIMAL: {
+                int precision = typeHandle.getRequiredColumnSize();
```

Comment:
Why did you remove check for 38 length? It may lead to query failure if Snowflake extends the max length, right?